opencensus/proto/trace/trace.proto

```diff
@@ -59,6 +61,26 @@ message Span {
   // This field is required.
   TruncatableString name = 4;
 
+  // Type of span. Can be used to specify additional relationships between spans
+  // in addition to a parent/child relationship.
+  enum SpanKind {
+    // Unspecified.
+    SPAN_KIND_UNSPECIFIED = 0;
+
+    // Indicates that the span covers server-side handling of an RPC or other
+    // remote network request.
+    SERVER = 1;
+
+    // Indicates that the span covers the client-side wrapper around an RPC or
+    // other remote request.
+    CLIENT = 2;
```

Comment:
I would add PRODUCER and CONSUMER as they are widely used, and indicate sending to a broker instead of directly to a server and visa versa

Reply:
OpenTracing also has [producer and consumer](https://github.com/opentracing/opentracing-java/blob/3e270abc6d07315ff71fe09886a9cbe93ea221f4/opentracing-api/src/main/java/io/opentracing/tag/Tags.java#L41) types. Might be good if we have that.